jcl/src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -256,9 +254,7 @@ void initReference (T r, ReferenceQueue q) {
 	 */
 	void dequeue() {
 		/*[PR 112508] not synchronized, so isEnqueued() could return wrong result */
```

Comment:
Perhaps this comment should be removed?

Reply:
I thought about removing it, but although I removed the synchronization, using volatile is still a fix for the same problem. In case it's some help in the future, I decided to leave it. It's a java preprocessor comment and is removed from the generated source code anyway.

Reply:
It only served to confuse before: there *was* synchronization even though the comment said otherwise. Even with synchronization `isEnqueued()` could only answer according to the state at that instant in time which may be wrong almost immediately. Now, without `synchronized`, it might be construed to mean that it's desirable that `isEnqueued()` can return the wrong result. Neither is good. In my view, bad comments are worse than no comments.

Reply:
The purpose of the comment is to indicate there was a bug related to this place in the code in the past, and to understand you should go and read the referenced RTC Problem Report.

Reply:
Removed the comment, not sure the referenced issue is still accessible.